README
These are informal and not-guaranteed-complete smapshots of some
previous releases of Reduce. There should not be any binaries or
serious build scripts here and these are all to be viewed as OUT
OF DATE and NOT SUPPORTED AT ALL. However some people may enjoy seeing
how the code-base has grown and getting a bit of insight into the
world of the past. And developers tracking a newly uncovered bug may
sometimes find it useful to look back into these archives in case
that gives insight.

PLEASE do not ask the main developers about building or installing
from these old files. ALL current support will be focussed on the
main version.

The files here typically have old restrictive copyright notices and
sometimes restrictive license terms. They are included here by virtue
of the permission that their originators granted to Tony Hearn and
his distributors to use them, but you should not modify and redistribute
anything from this directory without careful thought.


              Arthur Norman. August 2011



psl-1983/news.txt
28-Sep-82 17:50:20-PDT,3097;000000000000
Date: 28 Sep 1982 1750-PDT
From: Alan Snyder <AS>
Subject: new PSL!!!!
To: PSL-News: ;, PSL-Users: ;
cc: AS

Important Change to PSL!

We have installed a new version of PSL on HULK.  It contains a number of
significant changes which are described here.  In addition, you must change
your LOGIN.CMD file to TAKE PSL:LOGICAL-NAMES.CMD instead of
<PSL>LOGICAL-NAMES.CMD.  The <PSL> directory will disappear soon, so make this
change right away!

[These changes, except for NMODE, will appear on THOR and HEWEY shortly.  There
are no immediate plans to move NMODE to the Vax.]

Summary of changes:

* If you run "PSL", you will now get a PSL that contains the NMODE editor,
which is a replacement for EMODE.  PSL will start up in the editor, instead of
the PSL listen loop.  You can easily get back to the PSL listen loop from NMODE
by typing C-] L.  NMODE is a decent subset of EMACS, so if you are familiar
with EMACS you should be able to use NMODE without too much difficulty.  If you
are familiar with EMODE, you should read the file PSL:NMODE-GUIDE.TXT, which
explains the differences between NMODE and EMODE.  A printed copy of this memo,
including the NMODE command chart, is available in the documentation area next
to Helen Asakawa's office.

* The "PSL" program (what you get when you say "PSL" to EXEC) no longer
contains the PSL compiler.  Instead, there is a separate program for compiling
(Lisp) files.  To compile a file "FOO.SL", give the command "PSLCOMP FOO" to
EXEC.  PSLCOMP will produce a binary file "FOO.B" that can then be LOADed or
FASLINed.  To run the compiler interactively, just say "PSLCOMP" to EXEC.

* The PSL directories that contain the source and binaries for all PSL modules
have been moved to a private structure called SS: (the directories are now
SS:<PSL*>).  The old PSL directories (PS:<PSL*>) will disappear soon.  In
addition, the new directories have been reorganized somewhat to better reflect
the structure of the implementation.  The file PSL:-THIS-.DIRECTORY contains a
brief description of the new structure.  If you have used logical names to
refer to PSL directories, then this change should not cause too many problems.

* A number of small bug fixes and improvements have been made.  The most
notable improvements are (1) a more readable backtrace, (2) a better
prettyprinter, and (3) the definition of a "complete" set of I/O functions
taking an explicit channel argument (these functions all have names like
ChannelTerpri, where Terpri is an example of an I/O function that uses the
default I/O channels).  The file PSL:BUG-FIX.LOG contains an exhaustive listing
of the recent changes.

The documentation has been updated to reflect these changes.  The following new
or revised documents are available in the documentation area next to Helen
Asakawa's office:

	Notes on PSL at HP
	DEC-20 PSL New Users' Guide
	NMODE for EMODE Users
	How to customize NMODE

We have made "documentation packets" containing copies of these documents.
Users are encouraged to pick up a copy!
-------
11-Oct-82 15:55:41-PDT,5771;000000000000
Date: 11 Oct 1982 1555-PDT
From: Alan Snyder <AS>
Subject: new PSL installed
To: PSL-News: ;, PSL-Users: ;
cc: AS

PSL NEWS - 11 October 1982

A new PSL has been installed on Hulk and Hewey.  There are a number of
improvements, plus some INCOMPATIBLE changes (see below).  A most noticable
change (on Hulk) is that PSL no longer automatically starts up in the NMODE
editor.  However, if you want PSL to start up in the editor, you can still make
this happen using another new feature, INIT files (see below).  Otherwise, you
can explicitly enter NMODE by invoking the function NMODE, with no arguments.
In addtion, NMODE now supports the extended VT52 emulator on the 9836 (get the
latest version from Tracy).  (No, NMODE is not yet installed on Hewey.)

-------------------------------------------------------------------------------
INCOMPATIBLE CHANGES TO PSL:
-------------------------------------------------------------------------------
This latest version of PSL has 3 changes which may require some application
programs to be changed:

1. SAVESYSTEM

SaveSystem now takes 3 arguments.  The first argument is the banner, the second
is the file to be written, and the third is a list of forms to evaluated when
the new core image is started.  For example:

  (SaveSystem "PSL 3.1" "PSL.EXE" '((InitializeInterrupts)))

2. DUMPLISP

Dumplisp now takes 1 argument, the file to be written.  For example:

  (Dumplisp "PSL.EXE")

3. DSKIN

Dskin has been changed from a FEXPR to a single-argument EXPR.  This should
only affect calls to DSKIN with multiple arguments.  They will have to be
changed to several calls, each with one argument.

4. BR and UNBR

The functions BR and UNBR are no longer part of PSL.  These functions provided
a facility for breaking on entry and exit to specific functions.  However,
they didn't work very well and no one has figured out how to make them work,
so they have been removed.  Send complaints to PSL.

-------------------------------------------------------------------------------
MAJOR IMPROVEMENTS TO PSL:
-------------------------------------------------------------------------------
The following features have been added to PSL:

1. Init files

When PSL, RLISP, or PSLCOMP (note: not BARE-PSL) is executed, if a file
PSL.INIT, RLISP.INIT, or PSLCOMP.INIT, respectively, is in your home (login)
directory, it will be read and evaluated.  This allows you to automatically
customize your Lisp environment.  (The init files are .pslrc, .rlisprc, and
.pslcomprc on the Vax.) If you want PSL to come up in NMODE, include the
statement

  (setf nmode-auto-start T)

in your PSL.INIT file.

2. Prinlevel and Prinlength

The variables PRINLEVEL and PRINLENGTH now exist, as described in the Common
Lisp Reference Manual.  These variables allow you to limit the depth of

perq-pascal-lisp-project/exec.pas
procedure execute;
begin
if (code < 1) or (code > 201) then
    writeln('Illegal operation code: ', code)
else
case code of
    (* Case for DSKIN *)
1:    PAS350;
    (* Case for RDEVPR *)
2:    PAS349;
    (* Case for READCH *)
3:    PAS348;
    (* Case for LITER *)
4:    PAS347;
    (* Case for DIGIT *)
5:    PAS346;
    (* Case for UPBV *)
6:    PAS345;
    (* Case for PUTV *)
7:    PAS344;
    (* Case for MKVECT *)
8:    PAS343;
    (* Case for GETV *)
9:    PAS342;
    (* Case for UNFLUID *)
10:    PAS341;
    (* Case for GLOBALP *)
11:    PAS340;
    (* Case for GLOBAL *)
12:    PAS339;
    (* Case for REMOB *)
13:    PAS338;
    (* Case for INTERN *)
14:    PAS337;
    (* Case for FLUIDP *)
15:    PAS336;
    (* Case for VECTORP *)
16:    PAS335;
    (* Case for STRINGP *)
17:    PAS334;
    (* Case for FLOATP *)
18:    PAS333;
    (* Case for PRIN2TL *)
19:    PAS332;
    (* Case for FLUID *)
20:    PAS331;
    (* Case for PUTC *)
21:    PAS330;
    (* Case for MAPLIST *)
22:    PAS329;
    (* Case for MAPCON *)
23:    PAS328;
    (* Case for MAPCAR *)
24:    PAS327;
    (* Case for NCONC *)
25:    PAS326;
    (* Case for MAPCAN *)
26:    PAS325;
    (* Case for MAPC *)
27:    PAS324;
    (* Case for OR *)
28:    PAS323;
    (* Case for COND *)
29:    PAS322;
    (* Case for AND *)
30:    PAS321;
    (* Case for WHILE *)
31:    PAS320;
    (* Case for PROG *)
32:    PAS319;
    (* Case for MAP *)
33:    PAS318;
    (* Case for PROGG0131 *)
34:    PAS317;
    (* Case for GO *)
35:    PAS316;
    (* Case for RETURN *)
36:    PAS315;
    (* Case for PROG2 *)
37:    PAS314;
    (* Case for PROGN *)
38:    PAS313;
    (* Case for APPLY1 *)
39:    PAS312;
    (* Case for EVLIS *)
40:    PAS311;
    (* Case for DM *)
41:    PAS310;
    (* Case for DN *)
42:    PAS39;
    (* Case for DF *)
43:    PAS38;
    (* Case for DE *)
44:    PAS37;
    (* Case for PUTL *)
45:    PAS36;
    (* Case for REMD *)
46:    PAS35;
    (* Case for PUTD *)
47:    PAS34;
    (* Case for GETD *)
48:    PAS33;
    (* Case for TTHROW *)
49:    PAS32;
    (* Case for TCATCH *)
50:    PAS31;
    (* Case for RLIST *)
51:    PAS133;
    (* Case for EOFP *)
52:    PAS132;
    (* Case for REV *)
53:    PAS131;
    (* Case for REVX *)
54:    PAS130;
    (* Case for PRIN2 *)
55:    PAS129;
    (* Case for CDDDR *)
56:    PAS128;
    (* Case for CDDAR *)
57:    PAS127;
    (* Case for CDADR *)
58:    PAS126;
    (* Case for CDAAR *)
59:    PAS125;
    (* Case for CADDR *)
60:    PAS124;
    (* Case for CADAR *)
61:    PAS123;
    (* Case for CAADR *)
62:    PAS122;
    (* Case for CAAAR *)
63:    PAS121;
    (* Case for CDDDDR *)
64:    PAS120;
    (* Case for CDDDAR *)
65:    PAS119;
    (* Case for CDDADR *)
66:    PAS118;
    (* Case for CDDAAR *)
67:    PAS117;
    (* Case for CDADDR *)
68:    PAS116;
    (* Case for CDADAR *)
69:    PAS115;
    (* Case for CDAADR *)
70:    PAS114;
    (* Case for CDAAAR *)
71:    PAS113;
    (* Case for CADDDR *)
72:    PAS112;
    (* Case for CADDAR *)
73:    PAS111;
    (* Case for CADADR *)
74:    PAS110;
    (* Case for CADAAR *)
75:    PAS19;
    (* Case for CAADDR *)
76:    PAS18;
    (* Case for CAADAR *)
77:    PAS17;
    (* Case for CAAADR *)
78:    PAS16;
    (* Case for CAAAAR *)
79:    PAS15;
    (* Case for CDDR *)
80:    PAS14;
    (* Case for CDAR *)
81:    PAS13;
    (* Case for CADR *)
82:    PAS12;
    (* Case for CAAR *)
83:    PAS11;
    (* Case for *FIRST-PROCEDURE *)
84:    FIRSTP;
    (* Case for ADD1 *)
85:    XADD1;
    (* Case for XAPPLY *)
86:    XXAPPLY;
    (* Case for CAR *)
87:    XCAR;
    (* Case for CATCH *)
88:    XCATCH;
    (* Case for CDR *)
89:    XCDR;
    (* Case for CODEP *)
90:    XCODEP;
    (* Case for COMPRESS *)
91:    COMPRESS;
    (* Case for CONS *)
92:    XCONS;
    (* Case for CLOSE *)
93:    XCLOSE;
    (* Case for DIFFERENCE *)
94:    XDIFFERENCE;
    (* Case for DIVIDE *)
95:    XDIVIDE;
    (* Case for EVAL *)
96:    XEVAL;
    (* Case for EXPLODE *)
97:    EXPLODE;
    (* Case for FASTSTAT *)
98:    XFASTSTAT;
    (* Case for GENSYM *)
99:    GENSYM;
    (* Case for GREATERP *)
100:    XGREATERP;
    (* Case for LESSP *)
101:    XLESSP;
    (* Case for MINUS *)
102:    XMINUS;
    (* Case for NCONS *)
103:    XNCONS;
    (* Case for OPEN *)
104:    XOPEN;
    (* Case for PLUS2 *)
105:    XPLUS2;
    (* Case for PRINT *)
106:    XPRINT;
    (* Case for QUOTIENT *)
107:    XQUOTIENT;
    (* Case for RDTOK *)
108:    XRDTOK;
    (* Case for RDS *)
109:    XRDS;
    (* Case for READ *)
110:    XREAD;
    (* Case for RECLAIM *)
111:    XGCOLLECT;
    (* Case for REMAINDER *)
112:    XREMAINDER;
    (* Case for RPLACA *)
113:    XRPLACA;
    (* Case for RPLACD *)
114:    XRPLACD;
    (* Case for TERPRI *)
115:    XTERPRI;
    (* Case for TIMES2 *)
116:    XTIMES2;
    (* Case for THROW *)
117:    XTHROW;
    (* Case for UNBINDTO *)
118:    XUNBINDTO;
    (* Case for WRTOK *)
119:    XWRTOK;
    (* Case for WRS *)
120:    XWRS;
    (* Case for XCONS *)
121:    XXCONS;
    (* Case for PAIRP *)
122:    PAS21;
    (* Case for EQ *)
123:    PAS22;
    (* Case for NOT *)
124:    PAS23;
    (* Case for IDP *)
125:    PAS24;
    (* Case for NULL *)
126:    PAS25;
    (* Case for CONSTANTP *)
127:    PAS26;
    (* Case for EQN *)
128:    PAS27;
    (* Case for LIST2 *)
129:    PAS28;
    (* Case for LIST3 *)
130:    PAS29;
    (* Case for LIST4 *)
131:    PAS210;
    (* Case for LIST5 *)
132:    PAS211;
    (* Case for REVERSE *)
133:    PAS212;
    (* Case for APPEND *)
134:    PAS213;
    (* Case for MEMBER *)
135:    PAS214;
    (* Case for ERROR *)
136:    PAS215;
    (* Case for PAIR *)
137:    PAS216;
    (* Case for APPLY *)
138:    PAS217;
    (* Case for SASSOC *)
139:    PAS218;
    (* Case for ASSOC *)
140:    PAS219;
    (* Case for SUBLIS *)
141:    PAS220;
    (* Case for SUBST *)
142:    PAS221;
    (* Case for MEMQ *)
143:    PAS222;
    (* Case for ATSOC *)
144:    PAS223;
    (* Case for PUT *)
145:    PAS224;
    (* Case for DEFLIST *)
146:    PAS225;
    (* Case for EQUAL *)
147:    PAS226;
    (* Case for DELETE *)
148:    PAS227;
    (* Case for DELQ *)
149:    PAS228;
    (* Case for DELATQ *)
150:    PAS229;
    (* Case for GET *)
151:    PAS230;
    (* Case for REMPROP *)
152:    PAS231;
    (* Case for LENGTH *)
153:    PAS232;
    (* Case for ERRPRT *)
154:    PAS233;
    (* Case for MSGPRT *)
155:    PAS234;
    (* Case for FLAGP *)
156:    PAS235;
    (* Case for FLAG1 *)
157:    PAS236;
    (* Case for FLAG *)
158:    PAS237;
    (* Case for REMFLAG1 *)
159:    PAS238;
    (* Case for REMFLAG *)
160:    PAS239;
    (* Case for EQCAR *)
161:    PAS240;
    (* Case for PLIST *)
162:    PAS241;
    (* Case for VALUE *)
163:    PAS242;
    (* Case for FUNCELL *)
164:    PAS243;
    (* Case for SETPLIST *)
165:    PAS244;
    (* Case for SETVALUE *)
166:    PAS245;
    (* Case for SETFUNCELL *)
167:    PAS246;
    (* Case for ORDERP *)
168:    PAS247;
    (* Case for TOKEN *)
169:    PAS248;
    (* Case for ERRORSET *)
170:    PAS249;
    (* Case for NUMBERP *)
171:    PAS250;
    (* Case for FIXP *)
172:    PAS251;
    (* Case for MINUSP *)
173:    PAS252;
    (* Case for ABS *)
174:    PAS253;
    (* Case for SUB1 *)
175:    PAS254;
    (* Case for ZEROP *)
176:    PAS255;
    (* Case for ONEP *)
177:    PAS256;
    (* Case for EXPT *)
178:    PAS257;
    (* Case for FIX *)
179:    PAS258;
    (* Case for FLOAT *)
180:    PAS259;
    (* Case for EXPAND *)
181:    PAS260;
    (* Case for MAX *)
182:    PAS261;
    (* Case for MIN *)
183:    PAS262;
    (* Case for PLUS *)
184:    PAS263;
    (* Case for TIMES *)
185:    PAS264;
    (* Case for MAX2 *)
186:    PAS265;
    (* Case for MIN2 *)
187:    PAS266;
    (* Case for FUNCTION *)
188:    PAS267;
    (* Case for ATOM *)
189:    PAS268;
    (* Case for SET *)
190:    PAS269;
    (* Case for PRINC *)
191:    PAS270;
    (* Case for PRIN1 *)
192:    PAS271;
    (* Case for PRIN2T *)
193:    PAS272;
    (* Case for LBIND1 *)
194:    PAS273;
    (* Case for PBIND1 *)
195:    PAS274;
    (* Case for UNBIND1 *)
196:    PAS275;
    (* Case for UNBINDN *)
197:    PAS276;
    (* Case for LBINDN *)
198:    PAS277;
    (* Case for P.N *)
199:    PAS278;
    (* Case for EVLAM *)
200:    PAS279;
    (* Case for PBINDN *)
201:    PAS280;
end;  (* of case *)
end;
   